src/init/register_configure_agent.sh

```diff
@@ -8,74 +8,118 @@
+        ${sed} '/^$/d' "${file}"
+    else
+        unix_sed '/^$/d' "${file}"
+    fi
+}
+delete_auto_enrollment_tag() {
+    # Delete the configuration tag if its value is empty
+    # This will allow using the default value
+    if [ "${use_unix_sed}" = "False" ] ; then
+        ${sed} "s#.*<$1>.*</$1>.*##g" "${TMP_ENROLLMENT}"
+    else
+        unix_sed "s#.*<$1>.*</$1>.*##g" "${TMP_ENROLLMENT}"
+    fi
 
-    SET_ADDRESSES=("$@")
+    cat -s "${TMP_ENROLLMENT}" > "${TMP_ENROLLMENT}.tmp"
+    mv "${TMP_ENROLLMENT}.tmp" "${TMP_ENROLLMENT}"
+}
+
+# Change address block of the ossec.conf
+add_adress_block() {
+    # Getting function parameters on new variable
+    SET_ADDRESSES="$@"
 
     # Remove the server configuration
     if [ "${use_unix_sed}" = "False" ] ; then
-        ${sed} "/<server>/,/\/server>/d" ${INSTALLDIR}/etc/ossec.conf
+        ${sed} "/<server>/,/\/server>/d" ${CONF_FILE}
     else
-        unix_sed "/<server>/,/\/server>/d" "${INSTALLDIR}/etc/ossec.conf"
+        unix_sed "/<server>/,/\/server>/d" "${CONF_FILE}"
     fi
 
     # Get the client configuration generated by gen_ossec.sh
-    start_config="$(grep -n "<client>" ${INSTALLDIR}/etc/ossec.conf | cut -d':' -f 1)"
-    end_config="$(grep -n "</client>" ${INSTALLDIR}/etc/ossec.conf | cut -d':' -f 1)"
+    start_config="$(grep -n "<client>" ${CONF_FILE} | head -n 1 | cut -d':' -f 1)"
+    end_config="$(grep -n "</client>" ${CONF_FILE} | head -n 1 | cut -d':' -f 1)"
     start_config=$(( start_config + 1 ))
     end_config=$(( end_config - 1 ))
-    client_config="$(sed -n "${start_config},${end_config}p" ${INSTALLDIR}/etc/ossec.conf)"
+    client_config="$(sed -n "${start_config},${end_config}p" ${CONF_FILE})"
 
     # Remove the client configuration
     if [ "${use_unix_sed}" = "False" ] ; then
-        ${sed} "/<client>/,/\/client>/d" ${INSTALLDIR}/etc/ossec.conf
+        ${sed} "/<client>/,/\/client>/d" ${CONF_FILE}
     else
-        unix_sed "/<client>/,/\/client>/d" "${INSTALLDIR}/etc/ossec.conf"
+        unix_sed "/<client>/,/\/client>/d" "${CONF_FILE}"
     fi
 
     # Write the client configuration block
-    echo "<ossec_config>" >> ${INSTALLDIR}/etc/ossec.conf
-    echo "  <client>" >> ${INSTALLDIR}/etc/ossec.conf
-    for i in "${SET_ADDRESSES[@]}";
+    echo "<ossec_config>" >> ${CONF_FILE}
+    echo "  <client>" >> ${CONF_FILE}
+    for i in ${SET_ADDRESSES};
     do
-        echo "    <server>" >> ${INSTALLDIR}/etc/ossec.conf
-        echo "      <address>$i</address>" >> ${INSTALLDIR}/etc/ossec.conf
-        echo "      <port>1514</port>" >> ${INSTALLDIR}/etc/ossec.conf
-        echo "      <protocol>tcp</protocol>" >> ${INSTALLDIR}/etc/ossec.conf
-        echo "    </server>" >> ${INSTALLDIR}/etc/ossec.conf
+        echo "    <server>" >> ${CONF_FILE}
+        echo "      <address>$i</address>" >> ${CONF_FILE}
+        echo "      <port>1514</port>" >> ${CONF_FILE}
+        echo "      <protocol>udp</protocol>" >> ${CONF_FILE}
```

Comment:
Is there a reason to change this back to UDP?

Reply:
It should be tcp fixed her 941292d